spec/requests/profile_spec.rb

```diff
@@ -0,0 +1,164 @@
+        expect(response).to render_template(:index)
+      end
+
+      context "when profile belongs to logged in user" do
+        it "assigns profile instance to be that of logged in user" do
+          get profile_index_path, params: {uid: user1.uid}
+          expect(assigns(:profile)).to eq(user1)
+        end
+
+        context "when user has no moments and strategies" do
+          it "assigns stories instance with an empty array" do
+            get profile_index_path, params: {uid: user1.uid}
+            expect(assigns(:stories)).to match_array([])
+          end
+        end
+
+        context "when user has moments or strategies" do
+          it "assigns stories instance with user moments or strategies" do
+            stories = [build_stubbed(:moment), build_stubbed(:strategy)]
+            allow_any_instance_of(ProfileController)
+              .to receive(:get_stories).and_return(stories)
+            get profile_index_path, params: {uid: user1.uid}
+            expect(assigns(:stories)).to match_array(stories)
+          end
+        end
+      end
+
+      context "when profile does not belong to user" do
+        let(:profile_owner) { build_stubbed(:user2) }
+
+        before do
+          allow(User).to receive(:find_by).and_return(profile_owner)
+          get profile_index_path, params: {uid: profile_owner.uid}
+        end
+
+        it "does not assign stories" do
+          expect(assigns(:stories)).to be_nil
+        end
+
+        it "assigns profile instance to profile owner" do
+          expect(assigns(:profile)).to eq(profile_owner)
+        end
+      end
+    end
+  end
+
+  describe "#data" do
+    let!(:user) { create(:user) }
+    let!(:moment) { create(:moment, user: user) }
+
+    context "when the user is signed in" do
+      before { sign_in user }
+      before { get data_profile_index_path, params: {page: 1, uid: user.uid}, headers: {"ACCEPT" => "application/json"} }
+
+      it "returns a response with the correct path" do
+        expect(JSON.parse(response.body)["data"].first["link"]).to eq moment_path(moment)
+      end
+    end
+
+    context "when the user is not signed in" do
+      before { get data_profile_index_path params: {page: 1, uid: user.uid}, headers: {"ACCEPT" => "application/json"} }
+
+      it "returns a no_content status" do
+        expect(response).to have_http_status(:no_content)
+      end
+    end
+  end
+
+  describe "#add_ban" do
+    let(:user2) { create(:user2) }
+
+    before(:each) { Devise.mailer.deliveries.clear }
+
+    context "when admin does not exist" do
+      let(:user1) { create(:user1) }
+
+      it "cannot ban user" do
+        sign_in user1
+        post add_ban_profile_index_path, params: {user_id: user2.id}
+        expect(response.status).to eq(204)
+        expect(Devise.mailer.deliveries.count).to eq(0)
+      end
+    end
+
+    context "when admin exists" do
+      let(:user1) { create(:user1, admin: true) }
```

Comment:
```suggestion
      let(:admin_user) { create(:user1, admin: true) }
```

I think this would be a better variable name choice to know that this user is indeed the admin.